### DISCUSSION TITLE
Add documentation generator

### DISCUSSION THREAD

**Discussion by kevaundray** [noir-lang/noir#2710] [Ideas]
### Problem

Similar to rust-doc it would be helpful if Noir could have doc comments and a way for those to be exported such that they can be ingested by other tools.

### To-do

- Investigate the different ways to have something akin to rust-docs/doxygen

**Comment by Savio-Sou** [comment]
Let's convert this into Issue(s) when we can narrow it down into action item(s).

**Comment by Savio-Sou** [comment]
Converted into https://github.com/noir-lang/noir/issues/3362.
